### ISSUE TITLE
Improvement suggestion: Can't parse any errors

### ISSUE THREAD

**Issue by oej** [sbtinstruments/asyncio-mqtt#19]
The way you print the error as a string and return it makes it hard to parse. The example code tries reconnecting endlessly, even if the server is configured with bad port or host name.

Making the error parseable would be a huge improvement and would make it easy to improve the example code so it does not reconnect on any error, but on disconnects.

**Comment by frederikaalund** [comment]
Thanks for raising this issue. I'm open to changes. :) Feel free to make a pull request with your proposed changes.

**Comment by oej** [comment]
I am not that good in Python yet, but I realised that this is a problem that will stop production use of the library. I hope to get back and solve it together with you, but right now I'm stuck to parsing text strings which is not a reliable way forward. :-)

**Comment by frederikaalund** [comment]
Some code suggestions would be great! 👍

I'm not exactly sure I get the issue here. :) You don't need to parse any text. You can wrap your code in `try..except` and get the error object. What are you doing that requires you to parse error text?

**Comment by frederikaalund** [comment]
I'll close this issue for now due to inactivity. Let me know if you want to follow it up at a later point. :+1:
